### PR TITLE
Proposal - Deprecate ratpack-pac4j

### LINKED ISSUE #1414: Pac4j Doc Update
Update the manual to point to the external pac4j lib and deprecate the internal module.

### PR DESCRIPTION
**Pull request by michaelschlies**
fixes #1414 

With pac4j moving forward including multiple breaking changes the pac4j project has started maintaining their own instance of ratpack-pac4j. The ratpack project should deprecate its implementation.

---
This change is [Reviewable](https://reviewable.io/reviews/ratpack/ratpack/1422)

### CODE REVIEW

File: ratpack-pac4j/src/main/java/ratpack/pac4j/RatpackPac4j.java [hunk 1]
```diff
@@ -148,7 +150,7 @@ public static Handler authenticator(String path, ClientsProvider clientsProvider
    * This handler requires a {@link Clients} instance available in the context registry.
    * As such, this handler should be downstream of the {@link #authenticator(Client[])} handler.
    *
-   * <pre class="java">{@code
```
[the remark this answers is not in the document]
Comment (michaelschlies) [reply ratpack-pac4j/src/main/java/ratpack/pac4j/RatpackPac4j.java]: The Javadoc tests fail due to deprecation of the module, I can see if there is something we can do with the upstream dynamic test to not puke on deprecation if that is worthwhile.

[the remark this answers is not in the document]
Comment (michaelschlies) [reply ratpack-pac4j/src/main/java/ratpack/pac4j/RatpackPac4j.java]: Does it perhaps make sense to document that this module shouldn't be used in lieu of the pac4j version and actually deprecate the classes in 1.7 (per the below milestone) and remove with 2.0?

**Comment by johnrengelman** [comment]
@michaelschlies can you please rebase this to the latest master?

**Comment by johnrengelman** [comment]
Closing in favor of https://github.com/ratpack/ratpack/pull/1455

**Pull request closed without merging**